specs/language/introduction.tex

```diff
@@ -54,6 +54,47 @@
 where this document diverges this section, the remaining sections in this
 chapter, and the attached \ref{glossaries}.
 
+\Sec{Common Definitions}{Intro.Defs}
+
+\p The following definitions are consistent between \acrshort{hlsl} and the
+\gls{isoC} and \gls{isoCPP} specifications, however they are included here for
+reader convenience.
+
+\Sub{Diagnostic Message}{Intro.Defs.Diags}
+\p An implementation defined message belonging to a subset of the
+implementation's output messages which communicates diagnostic information to
+the user.
+
+\Sub{Ill-formed Program}{Intro.Defs.IllFormed}
+\p A program that is not well formed, for which the implementation is expected
+to return unsuccessfully and produce one or more diagnostic messages.
+
+\Sub{Implementation-defined Behavior}{Intro.Defs.ImpDef}
+\p Behavior of a well formed program and correct data which may vary by the
+implementation, and the implementation is expected to document the behavior.
+
+\Sub{Implementation Limits}{Intro.Defs.ImpLimits}
+\p Restrictions imposed upon programs by the implementation.
+
+\Sub{Undefined Behavior}{Intro.Defs.Undefined}
+\p Behavior of invalid program constructs or incorrect data which this standard
+imposes no requirements, or does not sufficiently detail.
+
+\Sub{Unspecified Behavior}{Intro.Defs.Unspecified}
+\p Behavior of a well formed program and correct data which may vary by the
+implementation, and the implementation is not expected to document the behavior.
+
+\Sub{Well-formed Program}{Intro.Defs.WellFormed} \p An \acrshort{hlsl} program
+constructed according to the syntax rules, diagnosable semantic rules, and the
+One Definition Rule.
```

Comment:
Interesting that ODR is called out as a separate rule, are there no other rules in the same classification (link-time) that need to be considered for a program to be considered "well-formed"?

Reply:
It's possible in the future we may need to incorporate more post-compile rules. I actually expect that we may accrue a few to represent the DXIL and SPIR-V validators, but I want to hold off referencing that explicitly until I get a little further along.